src/modeler/AutoRest.Swagger/Validation/AnonymousBodyParameter.cs

```diff
@@ -3,22 +3,29 @@
 
 using AutoRest.Core.Logging;
 using AutoRest.Core.Properties;
-using AutoRest.Swagger.Validation.Core;
 using AutoRest.Swagger.Model;
+using AutoRest.Swagger.Validation.Core;
 
 namespace AutoRest.Swagger.Validation
 {
-    public class AnonymousParameterTypes : TypedRule<SwaggerParameter>
+    public class AnonymousBodyParameter : TypedRule<SwaggerParameter>
```

Comment:
You may move `Id`, `ValidationCategory`, `Severity` & `MessageTemplate ` into top level `AnonymousTypeRule<T>` abstract class that would help in removing 13 for `new AvoidAnonymousTypes()`. upto you not required